python/tank/log.py

```diff
@@ -24,35 +25,100 @@ class LogManager(object):
     This class contains a series of methods to help standardize log output
     and access. Since Toolkit uses the standard python logging interface,
     you can manually configure and associate your logging if you like.
+
+    .. note:: This is a singleton class, so every time you instantiate it,
+              the same object is returned.
     """
-    # a global and standard rotating log file handler
-    # for writing generic toolkit logs to disk
-    _std_file_handler = None
 
-    @classmethod
-    def initialize_std_handler(cls, handler=logging.StreamHandler()):
+    # keeps track of the single instance of the class
+    __instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """
+        Ensures only one instance of the log manager exists.
+        """
+        # create the instance if it hasn't been created already
+        if not cls.__instance:
+            # remember the instance so that no more are created
+            instance = super(LogManager, cls).__new__(
+                cls,
+                *args,
+                **kwargs
+            )
+
+            # a global and standard rotating log file handler
+            # for writing generic toolkit logs to disk
+            instance._std_file_handler = None
+            instance._std_file_handler_log_name = None
+
+            # the root logger, created at code init
+            instance._root_logger = logging.getLogger(constants.ROOT_LOGGER_NAME)
+
+            # check the TK_DEBUG flag at startup
+            if constants.DEBUG_LOGGING_ENV_VAR in os.environ:
+                instance.global_debug = True
```

Comment:
This just made me think... how does this tie in with ```--debug``` on the command line?  Should that switch to triggering something here?